### PR TITLE
Add scenarios for OIDC Filter and Token Propagation

### PR DESCRIPTION
**Pull request by Sgitario**
Added some scenarios with the new OIDC Client Filter/Token propagation extensions.

### CODE REVIEW

**Comment by kshpak** [comment]
Fails against 1.11.0.Final, but succeed against master, which I think is right
@Sgitario can you please bump testcontainers version to 1.15.1, to fix some warnings that appear

**Comment by Sgitario** [comment]
> Fails against 1.11.0.Final, but succeed against master, which I think is right
> @Sgitario can you please bump testcontainers version to 1.15.1, to fix some warnings that appear

Yep, I was aware that the OIDC Filter extension is not working in 1.11.0.Final. It was fixed in the master as part of this commit: https://github.com/quarkusio/quarkus/commit/8a661fb8310f80c6560ba55193e5ebbbc0a843d4. 

About bumping testcontainers version, I would not like to do it as part of this PR as it will affect other modules as well and it's an unrelated change.

**Review by kshpak — APPROVED** [review]

**Comment by Sgitario** [comment]
@mjurc can you take a look into this PR?

**Comment by rsvoboda** [comment]
Rebase pleas ++ @mjurc take a look pls

**Comment by Sgitario** [comment]
> Rebase pleas ++ @mjurc take a look pls

Done ...

File: 013-quarkus-oidc-restclient/pom.xml [hunk 1]
```diff
@@ -34,6 +34,16 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-rest-client-mutiny</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-oidc-client-filter</artifactId>
+        </dependency>
+        <!-- Can't use both OIDC Client Filter and OIDC Token extensions.
```
Comment (mjurc) [inline 013-quarkus-oidc-restclient/pom.xml:41]: TODO would be convenient here

File: 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/ping/clients/TokenPropagationPongClient.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@RegisterRestClient
+// Can't use both OIDC Client Filter and OIDC Token extensions. Reported in https://github.com/quarkusio/quarkus/issues/14466
```
Comment (mjurc) [inline 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/ping/clients/TokenPropagationPongClient.java:11]: Again, the TODO would be really nice here

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/NativeTokenPropagationPingPongResourceIT.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.quarkus.qe;
+
+import org.junit.jupiter.api.Disabled;
+
+import io.quarkus.test.junit.NativeImageTest;
+
+@Disabled("Can't use both OIDC Client Filter and OIDC Token extensions. Reported in https://github.com/quarkusio/quarkus/issues/14466")
```
Comment (mjurc) [inline 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/NativeTokenPropagationPingPongResourceIT.java:7]: TODO again. It shows nicely in IDEs/editors :)

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/TokenPropagationPingPongResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package io.quarkus.qe;
+
+import org.junit.jupiter.api.Disabled;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@Disabled("Can't use both OIDC Client Filter and OIDC Token extensions. Reported in https://github.com/quarkusio/quarkus/issues/14466")
```
Comment (mjurc) [inline 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/TokenPropagationPingPongResourceTest.java:7]: One more TODO

**Review by mjurc — CHANGES_REQUESTED** [review]
Thanks for PR! Besides the nitpick though, this looks good to me.

File: 013-quarkus-oidc-restclient/pom.xml (hunk 1 above)
Comment (Sgitario) [reply 013-quarkus-oidc-restclient/pom.xml:41]: added

File: 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/ping/clients/TokenPropagationPongClient.java (hunk 2 above)
Comment (Sgitario) [reply 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/ping/clients/TokenPropagationPongClient.java:11]: added

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/NativeTokenPropagationPingPongResourceIT.java (hunk 3 above)
Comment (Sgitario) [reply 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/NativeTokenPropagationPingPongResourceIT.java:7]: added

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/TokenPropagationPingPongResourceTest.java (hunk 4 above)
Comment (Sgitario) [reply 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/TokenPropagationPingPongResourceTest.java:7]: added

**Review by mjurc — APPROVED** [review]

**Pull request merged**